python/ModelTools.py

```diff
@@ -278,12 +259,37 @@ def doNuisances(self):
             self.doObj("nuisancePdf", "PROD", ",".join(["%s_Pdf" % n for (n,nf,p,a,e) in self.DC.systs]))
             self.doSet("globalObservables", ",".join(globalobs))
 	
+    def doNuisancesGroups(self):
+        # Prepare a dictionary of which group a certain nuisance belongs to
+        groupsFor = {}
+        existingNuisanceNames = tuple(set([syst[0] for syst in self.DC.systs]+self.DC.flatParamNuisances.keys()))
```

Comment:
A separate loop is fine. The C++ part already handles the freezing of
categories so this could already be passed via --freezeNuisanceParameters
(the code just ignores categories set to frozen). We missed the chance to
group them so far however.


On Fri, Jul 10, 2015 at 1:07 PM, musella <notifications@github.com> wrote:

> In python/ModelTools.py
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/225#discussion_r34345727>
> :
>
> > @@ -278,12 +259,37 @@ def doNuisances(self):
> >              self.doObj("nuisancePdf", "PROD", ",".join(["%s_Pdf" % n for (n,nf,p,a,e) in self.DC.systs]))
> >              self.doSet("globalObservables", ",".join(globalobs))
> >  	
> > +    def doNuisancesGroups(self):
> > +        # Prepare a dictionary of which group a certain nuisance belongs to
> > +        groupsFor = {}
> > +        existingNuisanceNames = tuple(set([syst[0] for syst in self.DC.systs]+self.DC.flatParamNuisances.keys()))
>
> Hi, I can surely do that. Probably in a different loop since discrete
> nuisances are RooCategories and cannot be retrieve with ws.var. Also the
> code that does the freezing should RooCategories cannot be set const AFIUK
> so the freezing code should be aware of this. Is it the case? Cheers,
> Pasquale
> … <#14e77a6700795f55_>
> On 9 July 2015 at 16:36, André David <notifications@github.com> wrote: In
> python/ModelTools.py <
> https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/225#discussion_r34261524>
> : > @@ -278,12 +259,37 @@ def doNuisances(self): >
> self.doObj("nuisancePdf", "PROD", ",".join(["%s_Pdf" % n for (n,nf,p,a,e)
> in self.DC.systs])) > self.doSet("globalObservables", ",".join(globalobs))
> > > + def doNuisancesGroups(self): > + # Prepare a dictionary of which
> group a certain nuisance belongs to > + groupsFor = {} > +
> existingNuisanceNames = tuple(set([syst[0] for syst in
> self.DC.systs]+self.DC.flatParamNuisances.keys())) @nucleosynthesis
> <https://github.com/nucleosynthesis> <https://github.com/nucleosynthesis>
> proposed to also add self.DC.discretes to this expanded list. discretes is
> filled in
> https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/blob/slc6-root5.34.17/python/DatacardParser.py#L180-L183
> — Reply to this email directly or view it on GitHub <
> https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/225/files#r34261524>
> .
>  -- Pasquale Musella - ETHZ http://musella.web.cern.ch/musella
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/225/files#r34345727>
> .
>